api-admin/src/main/java/bio/terra/pearl/api/admin/service/ConfigExtService.java

```diff
@@ -39,4 +46,33 @@ private Map<String, String> buildConfigMap() {
         "adminApiHostname",
             StringUtils.defaultIfEmpty(applicationRoutingPaths.getAdminApiHostname(), ""));
   }
+
+  /**
+   * returns non-public configuration information -- note that this still should not return actual
+   * secrets
+   */
+  public Map<String, ?> getInternalConfigMap(AdminUser user) {
+    if (!user.isSuperuser()) {
+      throw new PermissionDeniedException("You do not have permission to view this config");
+    }
+    var configMap =
+        Map.of(
+            "pepperDsmConfig",
+            Map.of(
+                "useLiveDsm",
+                pepperDSMConfig.useLiveDsm(),
+                "secret",
+                pepperDSMConfig.getSecret().substring(0, 4)
+                    + "..."
+                    + pepperDSMConfig
+                        .getSecret()
+                        .substring(
+                            pepperDSMConfig.getSecret().length() - 3,
+                            pepperDSMConfig.getSecret().length()),
```

Comment:
This substringing is failing when I run the code, presumably because there is no secret?

Reply:
fixed to handle blank strings